server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/AllocationUtils.java

```diff
@@ -0,0 +1,365 @@
+/*******************************************************************************
+ * Copyright (c) 2021 University of Illinois and others.  All rights reserved.
+ * This program and the accompanying materials are made available under the
+ * terms of the Mozilla Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.mozilla.org/en-US/MPL/2.0/
+ *
+ *   Contributors:
+ *   Yong Wook Kim (NCSA) - initial API and implementation
+ *******************************************************************************/
+
+package edu.illinois.ncsa.incore.common.utils;
+
+import edu.illinois.ncsa.incore.common.auth.Authorizer;
+import edu.illinois.ncsa.incore.common.exceptions.IncoreHTTPException;
+import edu.illinois.ncsa.incore.common.models.Allocation;
+import edu.illinois.ncsa.incore.common.models.Space;
+import edu.illinois.ncsa.incore.common.models.SpaceUsage;
+import edu.illinois.ncsa.incore.common.dao.IAllocationRepository;
+import edu.illinois.ncsa.incore.common.dao.ISpaceRepository;
+import edu.illinois.ncsa.incore.common.dao.MongoSpaceDBRepository;
+import edu.illinois.ncsa.incore.common.dao.MongoAllocationDBRepository;
+import org.apache.log4j.Logger;
+
+
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.core.Response;
+
+public class AllocationUtils {
+
+    /***
+     * This method receives a username and to see if the user's number of dataset is within the allocation.
+     *
+     * @param allocationRepository IAllocationRepository injected by controller
+     * @param spaceRepository ISpaceRepository injected by controller
+     * @param username string representation of username
+     * @return postOk boolean if the user can post more datasets
+     */
+    public static Boolean checkNumDataset(IAllocationRepository allocationRepository,
```

Comment:
I suggest using better names for these methods that return booleans. checkNumDatasts, checkNumHazazardDatasets implies a numerical return. Consider something like:

canCreateHazardDataset, canCreateHazard, canCreateDataset

Reply:
changed like suggestion.